### PR TITLE
PLAT-100348: Migration Scroller to use hooks

### PR DESCRIPTION
**Pull request by 0x64**
Migration of `Scroller` component based on a hook migration of `@enact/ui`.

- Removes overscroll effect in code level
- Covers following prior PRs: #130, #139, #145, #154 , #169 , #175, #180, #183 

Known Issues:
- ~~Sometimes stuttering occurs after scrolling.~~ Fixed by ui's patch
- ~~This PR requires ui's patch (https://github.com/enactjs/enact/pull/2690)~~  Fixed by ui's patch

### CODE REVIEW

**Comment by ybsung** [comment]
LGTM

**Comment by ybsung** [comment]
https://github.com/enactjs/enact/pull/2690 PR should be merged first before this PR for the `dragging` setter.

**Comment by ryanjduffy** [comment]
Confirmed this resolves sampler build and test errors. Did notice some odd scroll bar indicator behavior in the Item QA sample including a duplicate native scroller inside the styled scroller.

**Comment by 0x64** [comment]
@ryanjduffy ,
`/stories/default/ItemKitchenSink.js` uses `agate/Scroller` and
`/stories/default/Icon.js` uses `ui/Scroller`.
And, we have `agate/Scroller` in `/src/AgateEnvironment/AgateEnvironment.js` also.
I don't know what happened behind the scene but it looks better to remove a scroller from `AgateEnvironment` at least, IMO. Could you please let me know what am I supposed to do to make samplers better?

**Comment by ryanjduffy** [comment]
I think you can remove the scrollers from the individual stories. I don't know the precise history but I expect they were oversights to add them to the stories.

File: Scrollable/ScrollButton.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ import Button from '../Button';
 import css from './Scrollbar.module.less';
 
 /**
- * A [Button]{@link agate/Button.Button} used within
+ * An [Button]{@link agate/Button.Button} used within
```
Comment (MikyungKim) [inline Scrollable/ScrollButton.js]: I don't think this is right ^^;;

Comment (0x64) [reply Scrollable/ScrollButton.js]: Good catch. :) Thanks!

File: Scrollable/ScrollButtons.js [hunk 2]
```diff
@@ -34,7 +32,7 @@ const
 	};
 
 /**
- * An Agate-styled scroll buttons. It is used in [Scrollbar]{@link agate/Scrollable.Scrollbar}.
+ * A Agate-styled scroll buttons. It is used in [Scrollbar]{@link agate/Scrollable.Scrollbar}.
```
Comment (MikyungKim) [inline Scrollable/ScrollButtons.js]: A -> An

Comment (0x64) [reply Scrollable/ScrollButtons.js]: Fixed this and more. Oops!

**Pull request merged**